Dockerfile

```diff
@@ -1,94 +1,88 @@
 FROM snapcore/snapcraft AS snapcraft
 # Using multi-stage dockerfile to obtain snapcraft binary
 
+FROM ubuntu:bionic AS flutterbuilder
+RUN apt-get update \
+    && apt-get install -y \
+        git curl unzip
+# Install Flutter from the beta channel
+RUN git clone --single-branch --depth=1 --branch beta https://github.com/flutter/flutter /opt/flutter 2>&1 \
+    && /opt/flutter/bin/flutter doctor -v
+
 FROM ubuntu:bionic AS xarbuilder
 RUN apt-get update \
 	&& apt-get install -y \
-		wget tar libssl1.0-dev libxml2-dev make g++
+		git libssl1.0-dev libxml2-dev make g++ autoconf
 RUN cd /tmp \
-	&& wget https://github.com/downloads/mackyle/xar/xar-1.6.1.tar.gz 2>&1 \
-	&& tar -zxvf xar-1.6.1.tar.gz 2>&1 \
-	&& mv xar-1.6.1 xar \
-	&& cd xar \
+    && git clone https://github.com/mackyle/xar 2>&1 \
+	&& cd xar/xar \
+	&& ./autogen.sh --noconfigure \
 	&& ./configure 2>&1 \
 	&& make 2>&1 \
 	&& make install 2>&1
 
-FROM dockercore/golang-cross:1.13.10 AS hover
+FROM ubuntu:bionic AS bomutilsbuilder
+RUN apt-get update \
+	&& apt-get install -y \
+	    git make g++
+RUN cd /tmp \
+	&& git clone https://github.com/hogliux/bomutils 2>&1 \
+	&& cd bomutils \
+	&& make 2>&1 \
+	&& make install 2>&1
 
-# Add dart apt repository
-RUN wget -qO- https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - 2>&1 \
-	&& wget -qO- https://storage.googleapis.com/download.dartlang.org/linux/debian/dart_stable.list > /etc/apt/sources.list.d/dart_stable.list
+# Fixed using https://github.com/AppImage/AppImageKit/issues/828
+FROM ubuntu:bionic as appimagebuilder
+RUN apt-get update \
+	&& apt-get install -y \
+	    curl
+RUN cd /opt \
+	&& curl -LO https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage \
+	&& chmod a+x appimagetool-x86_64.AppImage \
+	&& sed 's|AI\x02|\x00\x00\x00|g' -i appimagetool-x86_64.AppImage \
+	&& ./appimagetool-x86_64.AppImage --appimage-extract \
+	&& mv squashfs-root appimagetool
+
+FROM dockercore/golang-cross:1.13.10 AS hover
 
 # Install dependencies via apt
 RUN apt-get update \
-	&& apt-get upgrade -y \
 	&& apt-get install -y \
-		# why is gnupg-agent needed? Can this be removed??
-		gnupg-agent \
-		# for what command are these dependencies??
+	    # dependencies for flutter
+	    unzip \
```

Comment:
Is unzip still required in this stage since flutter moved into it's own stage?

Reply:
Nope